### DISCUSSION TITLE
Manage vimrc for windows and linux

### DISCUSSION THREAD

**Discussion by Dynge** [twpayne/chezmoi#1748] [Q&A] [answered]
Hello.

I want to manage my vimrc for both windows and linux machines. I would like to save my vimrc in "\~/vimfiles/" on my windows machine and "\~/.vim/" on my linux machine. Is there a way to use templates to achieve this on chezmoi?

**Comment by twpayne** [comment, marked as answer]
See [Handle different file locations on different systems with the same contents](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#handle-different-file-locations-on-different-systems-with-the-same-contents) in the how-to.

**Reply by Dynge** [reply]
Thank you! 
Sorry I didn't read the documentation properly. Poor skimming on my part. 😊

**Reply by j-xella** [reply]
> See [Handle different file locations on different systems with the same contents](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#handle-different-file-locations-on-different-systems-with-the-same-contents) in the how-to.

This is a solution for this particular problem, but I wonder if it is scalable enough? There is quite a lot of boilerplate code required for each file, and the amount of code is growing O(N^2) to the number of different places our file needs to be.

In reality, if we have an ocassional config file and a couple of cases (linux/windows), this is easily manageble. But what if we have a whole folder that needs to be in different places? Should we do the dance for each file in this folder, and then each time a new file is added there?

I wonder if it would be a good idea to extend chezmoi functionality with something like .chezmoiinclude file. If this file is found in source folder A and points to a source folder B, then the result is as if folder A has the contents of B. Then we could do something like this:

 * Put the whole real folder with config files somewhere like .hidden/app_config
 * Create linux/path/to/app_config/.chezmoiinclude(.hidden/app_config)
 * Create windows/path/to/app_config/.chezmoiinclude(.hidden/app_config)
 * Create required ignores for */path/to/app_config folders only

Or maybe there is already something like this in chezmoi?

**Reply by twpayne** [reply]
> Or maybe there is already something like this in chezmoi?

Yes, you can use symlinks to entire directories instead of files.
